sync-core/src/test/java/com/cloudant/sync/query/UnindexedMatcherTest.java

```diff
@@ -0,0 +1,394 @@
+        assertThat(UnindexedMatcher.compareEq(1.0, 1.0f), is(false));
+        assertThat(UnindexedMatcher.compareEq(1.0f, 1l), is(false));
+        assertThat(UnindexedMatcher.compareEq(1l, 1.0f), is(false));
+        assertThat(UnindexedMatcher.compareEq(1.0f, 1), is(false));
+        assertThat(UnindexedMatcher.compareEq(1, 1.0f), is(false));
+
+        assertThat(UnindexedMatcher.compareEq(1.0f, 1.1f), is(false));
+        assertThat(UnindexedMatcher.compareEq(1.1f, 1.0f), is(false));
+        assertThat(UnindexedMatcher.compareEq(1.1f, 1.0), is(false));
+        assertThat(UnindexedMatcher.compareEq(1.0, 1.1f), is(false));
+        assertThat(UnindexedMatcher.compareEq(1.1f, 1l), is(false));
+        assertThat(UnindexedMatcher.compareEq(1l, 1.1f), is(false));
+        assertThat(UnindexedMatcher.compareEq(1.1f, 1), is(false));
+        assertThat(UnindexedMatcher.compareEq(1, 1.1f), is(false));
+    }
+
+    @Test
+    public void compareEqStringToString() {
+        assertThat(UnindexedMatcher.compareEq("mike", "mike"), is(true));
+        assertThat(UnindexedMatcher.compareEq("mike", "Mike"), is(false));
+    }
+
+    @Test
+    public void compareEqStringToNumber() {
+        assertThat(UnindexedMatcher.compareEq("1", 1), is(false));
+        assertThat(UnindexedMatcher.compareEq(1, "1"), is(false));
+    }
+
+    @Test
+    public void compareEqDoubleToDouble() {
+        assertThat(UnindexedMatcher.compareEq(1.0, 1.0), is(true));
+        assertThat(UnindexedMatcher.compareEq(1.0, 1.00), is(true));
+        assertThat(UnindexedMatcher.compareEq(1.0, 1.1), is(false));
+        assertThat(UnindexedMatcher.compareEq(1.1, 1.0), is(false));
+    }
+
+    @Test
+    public void compareEqDoubleToLong() {
+        assertThat(UnindexedMatcher.compareEq(1.0, 1l), is(true));
+        assertThat(UnindexedMatcher.compareEq(1l, 1.0), is(true));
+        assertThat(UnindexedMatcher.compareEq(1.1, 1l), is(false));
+        assertThat(UnindexedMatcher.compareEq(1l, 1.1), is(false));
+    }
+
+    @Test
+    public void compareEqDoubleToInt() {
+        assertThat(UnindexedMatcher.compareEq(1.0, 1), is(true));
+        assertThat(UnindexedMatcher.compareEq(1, 1.0), is(true));
+        assertThat(UnindexedMatcher.compareEq(1.1, 1), is(false));
+        assertThat(UnindexedMatcher.compareEq(1, 1.1), is(false));
+    }
+
+    @Test
+    public void compareEqLongToLong() {
+        assertThat(UnindexedMatcher.compareEq(1l, 1l), is(true));
+        assertThat(UnindexedMatcher.compareEq(1l, 2l), is(false));
+        assertThat(UnindexedMatcher.compareEq(2l, 1l), is(false));
+    }
+
+    @Test
+    public void compareEqLongToInt() {
+        assertThat(UnindexedMatcher.compareEq(1l, 1), is(true));
+        assertThat(UnindexedMatcher.compareEq(1, 1l), is(true));
+        assertThat(UnindexedMatcher.compareEq(1l, 2), is(false));
+        assertThat(UnindexedMatcher.compareEq(2, 1l), is(false));
+    }
```

Comment:
Added UnindexMatcher.compareEq as a static import to the test class and added manual boxing to all Numbers being used during the compare equality tests.

https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-9845e00b021fbb4b1f9cd0cb43e43b23R15
https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-9845e00b021fbb4b1f9cd0cb43e43b23R61